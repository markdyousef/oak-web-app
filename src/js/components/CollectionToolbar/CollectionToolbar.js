// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import Dropdown from '../shared/Dropdown';
import colors from '../../styles/colors';
import { Box as Label, Chip } from '../shared/Label';

const Container = styled.div`
    max-width: 1050px;
    margin: 0 auto;
    display: flex;
    align-items: center;
    padding: 20px 20px 0;
    position: relative;
    @media (min-width: 1600px) {
        max-width: 1386px;
    }
`;

const ActionMenu = styled.div`
    font-size: 14px;
    margin-left: 10px;
    color: ${colors.grey};
    font-weight: normal;
    cursor: pointer;
    align-items: center;
    & span {
        color: ${colors.grey};
        margin-right: 4px;
        cursor: pointer;
    }
    & label {
        text-transform: capitalize;
        font-weight: bold;
        cursor: pointer;
    }
`;

const LabelsMenu = styled.div`
    font-size: 14px;
    margin-left: 16px;
    color: ${colors.grey};
    font-weight: normal;
    cursor: pointer;
    & span {
        color: ${colors.grey};
        margin-right: 4px;
        cursor: pointer;
    }
    & label {
        font-weight: bold;
        cursor: pointer;
    }
`;

const DropdownContainer = styled.div`
    position: absolute;
    top: 60px;
    z-index: 99;
    width: 200px;
`;

const MenuItem = styled.div`
    margin-bottom: 16px;
    cursor: pointer;
    font-weight: ${props => props.active ? 'bolder' : 'normal'};
    &:last-child {
        margin-bottom: 0;
    }
`;

const LabelContainer = styled.div`
    max-height: 150px;
    overflow: auto;
`;

const Labels = styled.div`
    display: flex;
    & div {
        margin-right: 4px;
    }
`;


type DefaultProps = {};
type Props = {
    active: string,
    onSort: () => void,
    labels: Array<Object>,
    filters: Array<string>,
    onFilter: (key:string) => void
};
type State = {
    showSort: bool,
    showFilter: bool
};

export default class Toolbar extends Component {
    static defaultProps: DefaultProps;
    props: Props;
    state: State;
    constructor() {
        super();
        this.state = {
            showSort: false,
            showFilter: false
        };
    }
    renderSortItems = () => {
        const { active, onSort } = this.props;
        const sortItems = [
            { name: 'Date' },
            { name: 'Comments' },
            { name: 'Likes' }
        ];

        return sortItems.map(item => (
            <MenuItem
                key={item.name}
                active={item.name === active}
                onClick={() => {
                    onSort(item.name);
                    this.setState({ showSort: false });
                }}
            >
                {item.name}
            </MenuItem>
        ));
    }
    renderFilterItems = () => {
        const { labels, filters, onFilter } = this.props;
        return (
            <LabelContainer>
                {labels.map((label) => {
                    const isActive = filters.indexOf(label.id) > -1;
                    return (
                        <Label
                            key={label.id}
                            onClick={() => onFilter(label.id)}
                            style={{ backgroundColor: label.color }}
                            name={label.name}
                            isActive={isActive}
                        />
                    );
                })}
            </LabelContainer>
        );
    }
    renderLabels = () => {
        const { labels, filters, onFilter } = this.props;
        if (labels) {
            return labels.map((label) => {
                const exists = filters.indexOf(label.id);
                if (exists > -1) {
                    return (
                        <Chip
                            key={label.id}
                            onClick={() => onFilter(label.id)}
                            style={{ backgroundColor: label.color }}
                            name={label.name}
                        />
                    );
                }
                return null;
            }).filter(Boolean);
        }
        return null;
    }
    render() {
        const { showSort, showFilter } = this.state;
        const { filters, active } = this.props;
        return (
            <Container>
                <div>
                    <ActionMenu onClick={() => this.setState({ showSort: !showSort })}>
                        <span>
                            Sort by:
                        </span>
                        <label>{active}</label>
                    </ActionMenu>
                    {showSort &&
                        <DropdownContainer>
                            <Dropdown
                                arrowPos="left"
                                onClose={() => this.setState({ showSort: false })}
                            >
                                {this.renderSortItems()}
                            </Dropdown>
                            </DropdownContainer>
                        }
                </div>
                <div>
                    <LabelsMenu onClick={() => this.setState({ showFilter: !showFilter })}>
                        <span>
                            Filter:
                        </span>
                        <label>{(filters && filters.length < 1) && 'Labels'}</label>
                    </LabelsMenu>
                    {showFilter &&
                        <DropdownContainer>
                            <Dropdown
                                arrowPos="left"
                                onClose={() => this.setState({ showFilter: false })}
                            >
                                {this.renderFilterItems()}
                            </Dropdown>
                            </DropdownContainer>
                        }
                </div>
                <Labels>
                    {this.renderLabels()}
                </Labels>
            </Container>
        );
    }
}
