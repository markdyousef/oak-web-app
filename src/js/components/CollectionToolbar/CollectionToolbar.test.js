import React from 'react';
import { shallow } from 'enzyme';
import CollectionToolbar from './CollectionToolbar';

it('renders correctly', () => {
    const shallowComponent = shallow(<CollectionToolbar />);
    expect(shallowComponent).toMatchSnapshot();
});
