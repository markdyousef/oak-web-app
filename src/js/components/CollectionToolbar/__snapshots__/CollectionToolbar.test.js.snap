// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <div>
        <styled.div
            onClick={[Function]}
        >
            <span>
                Sort by:
            </span>
            <label />
        </styled.div>
    </div>
    <div>
        <styled.div
            onClick={[Function]}
        >
            <span>
                Filter:
            </span>
            <label />
        </styled.div>
    </div>
    <styled.div />
</styled.div>,
  "nodes": Array [
    <styled.div>
      <div>
            <styled.div
                  onClick={[Function]}
            >
                  <span>
                        Sort by:
                  </span>
                  <label />
            </styled.div>
      </div>
      <div>
            <styled.div
                  onClick={[Function]}
            >
                  <span>
                        Filter:
                  </span>
                  <label />
            </styled.div>
      </div>
      <styled.div />
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Toolbar />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Toolbar {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "renderFilterItems": [Function],
        "renderLabels": [Function],
        "renderSortItems": [Function],
        "state": Object {
          "showFilter": false,
          "showSort": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <div>
                    <styled.div
                              onClick={[Function]}
                    >
                              <span>
                                        Sort by:
                              </span>
                              <label />
                    </styled.div>
          </div>
          <div>
                    <styled.div
                              onClick={[Function]}
                    >
                              <span>
                                        Filter:
                              </span>
                              <label />
                    </styled.div>
          </div>
          <styled.div />
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <div>
                    <styled.div
                              onClick={[Function]}
                    >
                              <span>
                                        Sort by:
                              </span>
                              <label />
                    </styled.div>
          </div>
          <div>
                    <styled.div
                              onClick={[Function]}
                    >
                              <span>
                                        Filter:
                              </span>
                              <label />
                    </styled.div>
          </div>
          <styled.div />
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Toolbar />,
}
`;
